staging/src/k8s.io/legacy-cloud-providers/gce/gce_instances.go

```diff
@@ -83,29 +83,82 @@ func (g *Cloud) ToInstanceReferences(zone string, instanceNames []string) (refs
 }
 
 // NodeAddresses is an implementation of Instances.NodeAddresses.
-func (g *Cloud) NodeAddresses(_ context.Context, _ types.NodeName) ([]v1.NodeAddress, error) {
-	internalIP, err := metadata.Get("instance/network-interfaces/0/ip")
-	if err != nil {
-		return nil, fmt.Errorf("couldn't get internal IP: %v", err)
+func (g *Cloud) NodeAddresses(ctx context.Context, nodeName types.NodeName) ([]v1.NodeAddress, error) {
+	timeoutCtx, cancel := context.WithTimeout(ctx, 1*time.Hour)
```

Comment:
What is the point of creating new context here?

Reply:
To pass to the GCE API, consistent with the NodeAddressesByProviderID function

Reply:
Still not sure the point of creating a new context with a 1 hour timeout. I agree we need to pass the context to the GCE API but we can just pass in the passed in context. The 1 hour timeout seems too long to be useful but if we want to do this we should probably make the timeout configurable.